### PR TITLE
Update free-programming-books-langs.md

### PR DESCRIPTION
**Pull request by erfatih**
[An Introduction to Statistical Learning with Applications in Python] link was changed. Existing link is dead.

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1940,7 +1940,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [A Guide to Python's Magic Methods](https://github.com/RafeKettler/magicmethods) - Rafe Kettler
 * [A Practical Introduction to Python Programming](https://www.brianheinold.net/python/) - Brian Heinold (HTML, PDF, Exercises sources)
 * [A Whirlwind Tour of Python](https://jakevdp.github.io/WhirlwindTourOfPython/) - Jake VanderPlas
-* [An Introduction to Statistical Learning with Applications in Python](https://hastie.su.domains/ISLP/ISLP_website.pdf) - Gareth James, Daniela Witten, Trevor Hastie, Robert Tibshirani, Jonathan Taylor (PDF)
+* [An Introduction to Statistical Learning with Applications in Python](https://www.statlearning.com/) - Gareth James, Daniela Witten, Trevor Hastie, Robert Tibshirani, Jonathan Taylor (PDF)
```
Comment (rng70) [inline books/free-programming-books-langs.md:1943]: trailing slash no longer accpeted. For further information see [here](https://github.com/vhf/remark-lint-no-url-trailing-slash#remark-lint-no-url-trailing-slash)

change
```text
https://www.statlearning.com/
```
to
```text
https://www.statlearning.com
```

**Review by rng70 — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (rng70) [reply books/free-programming-books-langs.md:1945]: unless you make the suggested change linter issue won't be solved

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1942,7 +1942,7 @@ Books on general-purpose programming that don't focus on a specific language are
...
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1945]:
```suggestion
* [An Introduction to Statistical Learning with Applications in Python](https://www.statlearning.com) - Gareth James, Daniela Witten, Trevor Hastie, Robert Tibshirani, Jonathan Taylor (PDF)
```

**Comment by eshellman** [comment]
added in #10988

**Pull request closed without merging**